### PR TITLE
feat: close toc on navigate on mobile

### PR DESCRIPTION
**Pull request by TomWoodward**

### CODE REVIEW

**Review by snm8 — APPROVED** [review]

File: src/app/content/components/ContentLink.tsx [hunk 1]
```diff
@@ -17,19 +17,23 @@ interface Props extends React.HTMLProps<HTMLAnchorElement> {
...
 // tslint:disable-next-line:variable-name
-export const ContentLink: SFC<Props> = ({book, page, currentPath, navigate, ...props}) => {
+export const ContentLink: SFC<Props> = ({book, page, currentPath, navigate, onClick, ...props}) => {
   const {url, params} = getBookPageUrlAndParams(book, page);
   const relativeUrl = toRelativeUrl(currentPath, url);
 
   return <a
     onClick={(e) => {
       e.preventDefault();
+      if (onClick) {
```
Comment (philschatz) [inline src/app/content/components/ContentLink.tsx]: Other code uses `onClick && onClick()` syntax; is there a reason for doing it differently here? Or is that just selective-memory?

File: src/app/content/components/Sidebar.tsx [hunk 2]
```diff
@@ -204,5 +211,8 @@ export default connect(
   (state: AppState) => ({
     ...selectors.bookAndPage(state),
     isOpen: selectors.tocOpen(state),
+  }),
+  (dispatch: Dispatch) => ({
+    onNavigate: () => dispatch(resetToc()),
```
Comment (philschatz) [inline src/app/content/components/Sidebar.tsx]: It seems that the ToC collapses _before_ the page changes; is that desired? It seems a bit confusing if the user is on a slow internet connection.

**Review by philschatz — CHANGES_REQUESTED** [review]
It seems that the ToC collapses _before_ the page changes; is that desired? It seems a bit confusing if the user is on a slow internet connection.

File: src/app/content/components/Sidebar.tsx (hunk 2 above)
Comment (TomWoodward) [reply src/app/content/components/Sidebar.tsx]: i think the ideal solution would be to add a loading state to the content side of the page. given that i think its better to close the toc immediately after clicking

File: src/app/content/components/ContentLink.tsx (hunk 1 above)
Comment (TomWoodward) [reply src/app/content/components/ContentLink.tsx]: the linter actually complains about doing that in a block. it fails the `no-unused-expression` rule. i think we might do that outside of blocks in `() => thing && thing()` expressions

File: src/app/content/components/Sidebar.tsx (hunk 2 above)
Comment (philschatz) [reply src/app/content/components/Sidebar.tsx]: @snm8 Are there plans to design a `Loading...` UI when the user navigates between pages? I can create a story if not. It's not critical but it is one of the weird intermediate states that rex can be in.

**Review by philschatz — APPROVED** [review]

**Comment by snm8** [comment]
Sure, we should have a standard one anyway that can be used across REX. Can you create another story for it, like you said?

**Pull request merged**